src/main/java/io/opentracing/contrib/global/GlobalSpanManager.java

```diff
@@ -0,0 +1,114 @@
+package io.opentracing.contrib.global;
+
+import io.opentracing.Span;
+
+import java.io.Closeable;
+import java.util.concurrent.Callable;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This class manages the globally <em>active</em> {@link Span}.<br>
+ * The default implementation will use a {@link ThreadLocal ThreadLocal storage} to maintain the active {@link Span}.
+ * <p>
+ * Each {@link Span} implementation obtained from the {@link GlobalTracer#tracer() global tracer} will automatically
+ * become the <em>active</em> {@link Span} until it is {@link Span#finish() finished} or {@link Span#close() closed}.<br>
+ * As a result, {@link io.opentracing.Tracer Tracer} clients need not worry about interacting with this
+ * {@link GlobalSpanManager} explicitly.
+ * <p>
+ * Providers of {@link io.opentracing.Tracer Tracer} implementations that want to customize the global Span management
+ * can provide their own
+ *
+ * @author Sjoerd Talsma
+ * @navassoc - activeSpan - io.opentracing.Span
+ */
+public abstract class GlobalSpanManager {
+    private static final Logger LOGGER = Logger.getLogger(GlobalSpanManager.class.getName());
+
+    /**
+     * Overridable singleton instance of the global span manager.
+     */
+    private static final AtomicReference<GlobalSpanManager> INSTANCE = new AtomicReference<GlobalSpanManager>();
```

Comment:
this can be way less fancy if the JVM/classloader owns the assignment. In that case you can use an enum singleton pattern. changing this at runtime is surprisingly advanced I think.